src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -1070,6 +1095,63 @@ public function withTimestamps($createdAt = null, $updatedAt = null)
         return $this->withPivot($this->createdAt(), $this->updatedAt());
     }
 
+    /**
+     * Specify that the pivot table has delete timestamp.
+     *
+     * @param  mixed  $deletedAt
+     * @return $this
+     */
+    public function withSoftDeletes($deletedAt = 'deleted_at')
+    {
+        $this->withSoftDeletes = true;
+
+        $this->pivotDeletedAt = $deletedAt;
+
+        $this->macro('withoutTrashedPivots', function () {
+            $this->query->withGlobalScope('withoutTrashed', function (Builder $query) {
+                $query->whereNull(
+                    $this->getQualifiedDeletedAtColumnName()
+                );
+            })->withoutGlobalScopes(['onlyTrashed']);
+
+            return $this;
+        });
+
+        $this->macro('withTrashedPivots', function () {
+            $this->query->withoutGlobalScopes(['withoutTrashed', 'onlyTrashed']);
+
+            return $this;
+        });
+
+        $this->macro('onlyTrashedPivots', function () {
+            $this->query->withGlobalScope('onlyTrashed', function (Builder $query) {
```

Comment:
Is this still correct since the methods have been renamed?

Reply:
renamed scopes